coderd/jwtutils/jwe.go

```diff
@@ -0,0 +1,127 @@
+package jwtutils
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/cryptokeys"
+)
+
+const (
+	encryptKeyAlgo     = jose.A256GCMKW
+	encryptContentAlgo = jose.A256GCM
+)
+
+// Encrypt encrypts a token and returns it as a string.
+func Encrypt(ctx context.Context, keys cryptokeys.Keycache, claims Claims) (string, error) {
+	signing, err := keys.Signing(ctx)
```

Comment:
One issue with this system is that you could call Encrypt with a key that's only meant for signing. It'd be nice if this method was `Encrypting` instead, and the underlying keycache rejected with an error if the key is being used for the wrong purpose